addon/components/sortable.js

```diff
@@ -0,0 +1,333 @@
+   @param {SortableItemModifier} item
+   */
+  @action
+  registerItem(item) {
+    if (this.items.indexOf(item) === -1) {
+      this.items = [...this.items, item];
+    }
+  }
+
+  /**
+   De-register an item with this group.
+   @method deregisterItem
+   @param {SortableItemModifier} item
+   */
+  @action
+  deregisterItem(item) {
+    const index = this.items.indexOf(item);
+    if (index !== -1) {
+      const items = [...this.items.slice(0, index), ...this.items.slice(index + 1)];
+      this.set("items", items);
+    }
+  }
+
+  /**
+   Register the group with this Sortable.
+   @method registerGroup
+   @param {SortableGroupModifier} group
+   */
+  @action
+  registerGroup(group) {
+    this._group = group;
+  }
+
+  /**
+   De-register the group with this Sortable.
+   @method deregisterGroup
+   @param {SortableGroupModifier} group
+   */
+  @action
+  deregisterGroup(group) {
+    if (this._group === group) {
+      this._group = null;
+    }
+  }
+
+  /**
+   Prepare for sorting.
+   Main purpose is to stash the current itemPosition so
+   we don’t incur expensive re-layouts.
+   @method _prepare
+   */
+  @action
+  prepare() {
+    this._itemPosition = this.itemPosition;
+  }
+
+  /**
+   Update item positions (relatively to the first element position).
+   @method update
+   */
+  @action
+  update() {
+    let sortedItems = this.sortedItems;
+    // Position of the first element
+    let position = this._itemPosition;
+
+    // Just in case we haven’t called prepare first.
+    if (position === undefined) {
+      position = this.itemPosition;
+    }
+
+    sortedItems.forEach(item => {
+      let dimension;
+      let direction = this.direction;
+
+      if (!item.isDragging) {
+        set(item, direction, position);
+      }
+
+      // add additional spacing around active element
+      if (item.isBusy) {
+        position += item.spacing * 2;
+      }
+
+      if (direction === 'x') {
+        dimension = 'width';
+      }
+      if (direction === 'y') {
+        dimension = 'height';
+      }
+
+      position += item[dimension];
+    });
+  }
+
+  /**
+   @method _commit
+   */
+  @action
+  commit() {
+    const items = this.sortedItems;
+    const itemModels = items.map(item => item.model);
+    const draggedItem = items.find(item => item.wasDropped);
+    let draggedModel;
+
+    if (draggedItem) {
+      draggedItem.wasDropped = false; // Reset
+      draggedModel = draggedItem.model;
+    }
+
+    this._updateItems();
+    this._onChange(itemModels, draggedModel);
+  }
+
+  @action
+  _onChange(itemModels, draggedModel) {
+    const groupModel = this.groupModel;
+
+    if (groupModel !== NO_MODEL) {
+      this.onChange(groupModel, itemModels, draggedModel);
+    } else {
+      this.onChange(itemModels, draggedModel);
+    }
+  }
+
+  /**
+   * API of methods needed by the modifiers
+   *
+   * This api was created to prevent re-rendering when some of the things change. These things are needec
```

Comment:
nit: `needec` -> `needed`

Reply:
thanks, fixed. Got a couple other minor ones I found. Will commit and push later. Wait to see if you find anymore